src/core/ColorPicker/ColorPicker.tsx

```diff
@@ -12,14 +12,29 @@ import {
   useEventListener,
 } from '../utils';
 import cx from 'classnames';
-import { ColorValue, HsvColor } from '../utils/color/ColorValue';
+import { ColorType, ColorValue, HsvColor } from '../utils/color/ColorValue';
 import { Slider } from '../Slider';
 import ColorSwatch from './ColorSwatch';
 import { IconButton } from '../Buttons';
 import { Input } from '../Input';
 
 // cspell: ignore borderless tbgr
 
+export const getColorValue = (color: ColorType | ColorValue | undefined) => {
+  if (color instanceof ColorValue) {
+    return color;
+  }
+  if (undefined === color) {
+    return ColorValue.fromTbgr(0);
+  }
+
+  try {
+    return ColorValue.create(color);
+  } catch (_e) {
+    return ColorValue.fromTbgr(0);
+  }
```

Comment:
Is all this validation really needed? ColorValue.create should already return a default color if it fails to parse the input

Reply:
Yes, because ColorValue.fromString will now throw instead of returning a default color. This was needed in the hex input to know if the user put in a string that can't be converted.

Reply:
I could remove the validation here if a make ColorValue.fromString public again so it could be called when the user is typing in a value and it cannot be converted the value would not automatically jump to a default color value.... they may have just have mistyped one character.  I could then add `| undefined` to `ColorType` so .create would just return a default black color, then this function could be removed.